quic/s2n-quic-core/src/dc/disabled.rs

```diff
@@ -0,0 +1,37 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    crypto::tls::TlsSession,
+    dc::{ConnectionInfo, Endpoint, Path},
+    stateless_reset::Token,
+};
+
+#[derive(Debug, Default)]
+pub struct Disabled(());
+
+impl Endpoint for Disabled {
+    const ENABLED: bool = false;
+
+    type Path = DisabledPath;
+
+    fn new_path(&mut self, _connection_info: &ConnectionInfo) -> Self::Path {
+        DisabledPath(())
+    }
+}
+
+pub struct DisabledPath(());
+
+impl Path for DisabledPath {
+    fn on_path_secrets_ready(&mut self, _session: &impl TlsSession) {}
```

Comment:
Will any of these methods actually get called? Might be a good idea to put an `unimplemented!()` or at the very least a `debug_assert!(false)`.

Reply:
If I make `new_path` return Option<Path>, then I can just have the disabled endpoint return None, in which case these will not be called